src/platform/src/main/java/org/geoserver/platform/resource/Resources.java

```diff
@@ -121,6 +121,38 @@ public static File find(Resource resource) {
         }
     }
 
+    /**
+     * Checks {@link Resource#getType()} and returns existing file() or dir() as appropriate, or
+     * null for {@link Resource.Type#UNDEFINED}.
+     *
+     * <p>This approach is a reproduction of GeoServerResourceLoader find logic.
+     *
+     * @see Resource#dir()
+     * @see Resource#file()
+     * @param resource Resource indicated
+     * @return The file if exists, null if {@link Resource.Type#UNDEFINED} and force is false, a
+     *     File with the resource path otherwise
+     */
+    public static File find(Resource resource, boolean force) {
```

Comment:
Maybe more appropriate use call this new method "file" rather than find (since it is not really a search).

Reply:
The original method (without force parameter) is called find, why do you want to call this one file instead? I just needed a way to avoid a file being searched without forcing it to be materialized on disk if not present.

Reply:
Holding this chat open so I am reminded to javadoc `force`.

Reply:
Tried to document it, but not enough I guess:

```
@return The file if exists, null if {@link Resource.Type#UNDEFINED} and force is false, a
     File with the resource path otherwise
``

Reply:
No it is good, I get the use case now.